contracts/test/branded_token/MockBrandedTokenFail.sol

```diff
@@ -0,0 +1,125 @@
+pragma solidity ^0.5.0;
+
+// Copyright 2018 OpenST Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import "./../../BrandedToken.sol";
+
+
+/**
+ * @title MockBrandedTokenFail.
+ *
+ * @notice Supports testing of BrandedToken(BT) failure cases.
+ */
+contract MockBrandedTokenFail is BrandedToken {
```

Comment:
I do not believe this test need to inherit from or import the genuine BT contract; and, similarly, likely does not need the constructor.

However, the above comments may lose some sensibility:
- if you make changes to use of BT for positive tests, as a code/typing/prioritization/efficiency matter; and/or
- if you add a validation to confirm that the `valueToken` for GC and BT match

Reply:
One of the reason for MockBrandedTokenFail to inherit BrandedToken is usage of convertToBrandedTokens. GC in requestStake validates mintBT amount by checking to brandedToken.convertToBrandedTokens. I wanted to avoid duplication of actual implementation of convertToBrandedTokens in MockBrandedTokenFail.

```if you make changes to use of BT for positive tests, as a code/typing/prioritization/efficiency matter; and/or```
BT is being used for positive tests.

```if you add a validation to confirm that the valueToken for GC and BT match```
Validation has been added.